crates/router/src/connector/adyen/transformers.rs

```diff
@@ -3325,14 +3297,22 @@ pub fn get_qr_metadata(
     let image_data_url = Url::parse(image_data.data.as_str())
         .ok()
         .ok_or(errors::ConnectorError::ResponseHandlingFailed)?;
+    let qr_code_url = response
+        .action
+        .qr_code_url
+        .clone()
+        .ok_or(errors::ConnectorError::ResponseHandlingFailed)?;
 
-    let qr_code_instructions = payments::QrCodeNextStepsInstruction {
-        image_data_url,
-        display_to_timestamp: None,
-    };
+    let qr_code_instructions = payments::NextActionFromConnectorMetaData::QrCodeInformation(
+        payments::QrCodeInformation::QrCodeUrl {
```

Comment:
Based on the data available, please construct QrCodeInformation::QrCodeUrl or QrCodeInformation::QrDataUrl or QrCodeInformation::QrCodeImageUrl.